client/landing/stepper/declarative-flow/migration-signup.ts

```diff
@@ -50,61 +49,24 @@ const migrationSignup: Flow = {
 
 		let result: AssertConditionResult = { state: AssertConditionState.SUCCESS };
 
-		const flowName = this.name;
-
-		const locale = useFlowLocale();
-
 		const queryParams = new URLSearchParams( window.location.search );
-		const aff = queryParams.get( 'aff' );
-		const vendorId = queryParams.get( 'vid' );
-		const ref = queryParams.get( 'ref' );
-
-		const getStartUrl = () => {
-			let hasFlowParams = false;
-			const flowParams = new URLSearchParams();
-			const queryParams = new URLSearchParams();
-
-			if ( vendorId ) {
-				queryParams.set( 'vid', vendorId );
+		const loginUrlParams: Record< string, string > = {};
+		for ( const [ key, value ] of queryParams.entries() ) {
+			if ( [ 'aff', 'vid', 'ref' ].includes( key ) && value ) {
+				loginUrlParams[ key ] = value;
 			}
+		}
 
-			if ( aff ) {
-				queryParams.set( 'aff', aff );
-			}
-
-			if ( ref ) {
-				queryParams.set( 'ref', ref );
-			}
-
-			if ( locale && locale !== 'en' ) {
-				flowParams.set( 'locale', locale );
-				hasFlowParams = true;
-			}
-
-			const redirectTarget =
-				`/setup/${ FLOW_NAME }` +
-				( hasFlowParams ? encodeURIComponent( '?' + flowParams.toString() ) : '' );
-
-			let queryString = `redirect_to=${ redirectTarget }`;
-
-			if ( queryParams.toString() ) {
-				queryString = `${ queryString }&${ queryParams.toString() }`;
-			}
-
-			const logInUrl = getLoginUrl( {
-				variationName: flowName,
-				locale,
-			} );
-
-			return `${ logInUrl }&${ queryString }`;
-		};
+		const logInUrl = useFlowLoginUrl( {
+			flow: this,
```

Comment:
I think we should not pass the full flow here because the flow object is pretty complex and can encourage other developers to start to use the flow hooks inside this hook.

Reply:
That's a fair comment. One reason I did this is because the URL path include either the flow name (`flow.name`) _or_ the variation name (`flow.variationSlug`). The caller shouldn't need to know that they need to use `flow.variationSlug ?? flow.name` here -- I am confident we have some bugs around this already, despite there being a small number of flow variations out there.

Reply:
I have tackled this as follows in 96454ffdd8683c904ffdb6f47706c8f8852e61c8:
* The code in `useFlowLoginUrl()` now destructures the `flow` argument so we're only referencing `flow.name` and `flow.variantSlug`
* I've added a comment in `useFlowLoginUrl()` to warn against using hooks on the `flow` object.

This won't stop anyone from doing it wrong, but I think we're likely to see many more people touching the calling code over time than we are to see people modifying the implementation details in this hook.